internal/version/release_version.go

```diff
@@ -18,18 +18,19 @@ import (
 	"context"
 	"time"
 
-	"github.com/google/go-github/v38/github"
+	"github.com/google/go-github/v42/github"
 )
 
 //go:generate mockgen -destination=../mocks/mock_release_version.go -package=mocks github.com/mongodb/mongocli/internal/version ReleaseVersionDescriber
 
-type ReleaseVersionDescriber interface {
-	LatestVersion() (*ReleaseInformation, error)
-}
+const (
+	maxWaitTime = 500 * time.Millisecond
+)
 
-type ReleaseInformation struct {
-	Version     string
-	PublishedAt time.Time
+type CriteriaType func(tag string, tool string) bool
```

Comment:
[nit] there's no need to call things what they are, ie, you would not say `nameString`
```suggestion
type Criteria func(string, string) bool
```